ambry-tools/src/main/java/com.github.ambry/store/DumpData.java

```diff
@@ -667,20 +651,39 @@ private void compareIndexEntriesToLogContent(File indexFile, File replicaDirecto
               blobValue.getSize() + " Original Message Offset " + blobValue.getOriginalMessageOffset() +
               " Flag " + blobValue.getFlags() + "\n";
           boolean isDeleted = blobValue.isFlagSet(IndexValue.Flags.Delete_Index);
-          if (deletedMsgInfo != null && isDeleted) {
-            deletedMsgInfo.put(blobValue.getOriginalMessageOffset(), blobValue.getOffset());
-          }
           boolean success =
               dumpDataHelper.readFromLogAndVerify(randomAccessFile, blobValue.getOffset(), key.getID(), blobValue,
-                  coveredRangesInLog);
+                  coveredRanges);
           if (!success) {
             logger.error("Failed for Index Entry " + msg);
           }
+          if (isDeleted) {
+            long originalOffset = blobValue.getOriginalMessageOffset();
+            deleteMsgsOriginalOffsets.add(originalOffset);
+            if (originalOffset != -1 && !coveredRanges.containsKey(originalOffset) &&
```

Comment:
do you need the `originalOffset != 1` condition? Won't the `originalOffset >= startOffset` take care of this ?